### DISCUSSION TITLE
How can I restore the conversation after reboot the code by using official api?

### DISCUSSION THREAD

**Discussion by chun1617** [acheong08/ChatGPT#521] [General]
Hello! It is working with official api quite well.
But it will reset the whole conversation if reboot the code.
How can I make it using the same conversation even reboot the code?

**Comment by acheong08** [comment]
I added some functions for saving conversations locally. Haven't tested. Code only, not for CLI. Check https://github.com/acheong08/ChatGPT/wiki/revChatGPT#dump_conversation_history and read some code. I'll need some time to properly implement and test this

**Reply by chun1617** [reply]
yeah i tried this also, but it look like will save every message as one json. and also look like hard to load the conversation

**Reply by acheong08** [reply]
Bad coding practices on my part. I'll change the implementation so don't count on it yet

**Reply by acheong08** [reply]
When it is ready, it will go into CLI which is when you know it is usable

**Reply by chun1617** [reply]
okay thank you!

**Reply by acheong08** [reply]
@chun1617 Not well documented but please check out the code. Also implemented in CLI

**Reply by chun1617** [reply]
i tried to use chatbot.conversations.load("1071493713815158964.json") to restore the conversations but look like chatgpt doesn't got the file.

**Reply by acheong08** [reply]
You have to save it first

**Reply by chun1617** [reply]
Yes I saved it with this then try to load the conversation after reboot the bot.
```
chatbot.save_conversation(response['id'])
filename = "1071493713815158964.json"
chatbot.conversations.save(filename)
```

**Reply by acheong08** [reply]
Save conversation to memory: `chatbot.save_conversation(...)`
Save conversation from memory to file: `chatbot.conversations.save(...)`
Load conversation from file to memory: `chatbot.conversations.load(...)`
Load conversation from memory to active conversation: `chatbot.load_conversation(...)`

**Reply by acheong08** [reply]
4 step process

**Reply by chun1617** [reply]
Yes its work now! Thank you!

**Comment by chun1617** [comment]
and do you have an idea to resolve this problem?
```ValueError: Encountered text corresponding to disallowed special token '<|endoftext|>'.
If you want this text to be encoded as a special token, pass it to `allowed_special`, e.g. `allowed_special={'<|endoftext|>', ...}`.
If you want this text to be encoded as normal text, disable the check for this token by passing `disallowed_special=(enc.special_tokens_set - {'<|endoftext|>'})`.
To disable this check for all special tokens, pass `disallowed_special=()`.```

**Reply by acheong08** [reply]
`openai.error.InvalidRequestError: Unrecognized request argument supplied: disallowed_special`

Might be a new feature not part of OpenAI's API yet

**Reply by acheong08** [reply]
The API is down right now so can't test

**Reply by acheong08** [reply]
Can't get the error
